Lib/test/test_capi/test_sys.py

```diff
@@ -211,6 +206,74 @@ def test_sys_writestderr(self):
         # Test PySys_WriteStderr()
         self._test_sys_writestream('PySys_WriteStderr', 'stderr')
 
+    @unittest.skipIf(_testlimitedcapi is None, 'need _testlimitedcapi module')
+    def test_sys_audit(self):
+        # Test PySys_Audit()
+        sys_audit = _testlimitedcapi.sys_audit
+        audit_events = []
+        def audit_hook(event, args):
+            audit_events.append((event, args))
+            return None
+
+        sys.addaudithook(audit_hook)
+        try:
+            result = sys_audit("test.event", "OO", 1, "a")
+            self.assertEqual(result, 0)
+            self.assertEqual(audit_events[-1][0], "test.event")
+            self.assertEqual(audit_events[-1][1], (1, "a"))
+
+            result = sys_audit("test.no_args", "")
+            self.assertEqual(result, 0)
+            self.assertEqual(audit_events[-1][0], "test.no_args")
+            self.assertEqual(audit_events[-1][1], ())
+
+            with self.assertRaises(TypeError):
+                sys_audit(123, "O", 1)
+
+            result = sys_audit("テスト.イベント", "O", 42)
+            self.assertEqual(result, 0)
+            self.assertEqual(audit_events[-1][0], "テスト.イベント")
+            self.assertEqual(audit_events[-1][1], (42,))
+
+            with self.assertRaises(UnicodeDecodeError):
+                sys_audit(b"test.non_utf8\xff", "O", 1)
+        finally:
+            sys.audit_hooks = []
+
+    @unittest.skipIf(_testlimitedcapi is None, 'need _testlimitedcapi module')
+    def test_sys_audittuple(self):
+        # Test PySys_AuditTuple()
+        sys_audittuple = _testlimitedcapi.sys_audittuple
+        audit_events = []
+        def audit_hook(event, args):
+            audit_events.append((event, args))
+            return None
+
+        sys.addaudithook(audit_hook)
+        try:
+            result = sys_audittuple("test.event", (1, "a"))
+            self.assertEqual(result, 0)
+            self.assertEqual(audit_events[-1][0], "test.event")
+            self.assertEqual(audit_events[-1][1], (1, "a"))
+
+            result = sys_audittuple("test.null_tuple")
+            self.assertEqual(result, 0)
+            self.assertEqual(audit_events[-1][0], "test.null_tuple")
+            self.assertEqual(audit_events[-1][1], ())
+
+            with self.assertRaises(TypeError):
+                sys_audittuple("test.bad_tuple", [1, 2])
+
+            result = sys_audittuple("テスト.イベント", (42,))
+            self.assertEqual(result, 0)
+            self.assertEqual(audit_events[-1][0], "テスト.イベント")
+            self.assertEqual(audit_events[-1][1], (42,))
+
+            with self.assertRaises(UnicodeDecodeError):
+                sys_audittuple(b"test.non_utf8\xff", (1,))
+        finally:
```

Comment:
Add also test for `sys_audittuple(NULL, (1,))`. If it crashes, just add a comment, like in other tests above.